src/Cake.Core.Tests/Unit/Scripting/Processors/LoadDirectiveProcessorTests.cs

```diff
@@ -137,5 +138,41 @@ public void Should_Insert_Line_Directives_When_Processing_Load_Directives()
             Assert.Equal(result.Lines[11], "// #l b.cake");
             Assert.Equal(result.Lines[12], "int y=2;");
         }
+
+        [Fact]
+        public void Should_Process_AbsolutePath_Script_Reference_Found_In_Source_LegacyFormat()
+        {
+            // Given
+            var fixture = new ScriptAnalyzerFixture();
+            var absPath = Directory.GetCurrentDirectory().Replace('\\', '/');
+            fixture.Providers.Add(new FileLoadDirectiveProvider());
+            fixture.GivenScriptExist("/Working/script.cake", $"#load \"{absPath}/utils.cake\"");
+            fixture.GivenScriptExist($"{absPath}/utils.cake", "Console.WriteLine();");
+
+            // When
+            var result = fixture.Analyze("/Working/script.cake");
+
+            // Then
+            Assert.Equal(1, result.Script.Includes.Count);
+            Assert.Equal($"{absPath}/utils.cake", result.Script.Includes[0].Path.FullPath);
+        }
+
+        [Fact]
+        public void Should_Process_AbsolutePath_Script_Reference_Found_In_Source()
+        {
+            // Given
+            var fixture = new ScriptAnalyzerFixture();
+            var absPath = Directory.GetCurrentDirectory().Replace('\\', '/');
```

Comment:
Don't use System.IO here and don't get the user's current directory.

The test fixture is using a fake filesystem, so simply declare an absolute path like this `/Absolute/utils.cake`.

If you want to test Windows paths, you will need to create a Windows specific test with `WindowsFact` or `WindowsTheory` and declare an absolute path like this `C:/utils.cake`.

Reply:
Done